src/config.rs

```diff
@@ -114,7 +114,41 @@ pub struct Builder {
     pub from_name: String,
     pub from_address: Option<String>,
     pub smtp_server: Option<String>,
-    pub providers: HashMap<String, Provider>,
+    pub providers: HashMap<String, ProviderBuilder>,
```

Comment:
I don't quite get the goal of this stuff. Why do we need the incrementalism? Would make for a good addition to the commit message, too.